src/components/MainForm/Mainform.tsx

```diff
@@ -67,152 +159,178 @@ export default function Mainform() {
   } = useForm<IFormInput>();
 
   const FormSubmitHandler: SubmitHandler<IFormInput> = (data: IFormInput) => {
+    history.push("./loading");
     console.log(data);
+    store.dispatch({
+      type: "ADD_DATA",
+      payload: data,
+    });
   };
   const [Fname, setFName] = useState("");
   const [Lname, setLName] = useState("");
```

Comment:
Why we need these states , when we are already storing them in Form

Reply:
Also use camel case for states like fName

Reply:
These states removed